### PR TITLE
[SPARK-43213][PYTHON] Add `DataFrame.offset` to vanilla PySpark

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Add `DataFrame.offset` to PySpark

### Why are the changes needed?
`DataFrame.offset` was supported in Scala side and Spark Connect since 3.4, but it is missing in vanilla PySpark.

### Does this PR introduce _any_ user-facing change?
yes, new API

### How was this patch tested?
added doctests

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @HyukjinKwon @xinrong-meng

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -1273,6 +1273,41 @@ def limit(self, num: int) -> "DataFrame":
         jdf = self._jdf.limit(num)
         return DataFrame(jdf, self.sparkSession)
 
+    def offset(self, num: int) -> "DataFrame":
```
Comment (HyukjinKwon) [inline python/pyspark/sql/dataframe.py:1276]: Should also add to the Python API reference .rst file I believe.

Comment (zhengruifeng) [reply python/pyspark/sql/dataframe.py:1276]: good catch! thanks!

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**